orderer/common/filerepo/filerepo.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package filerepo
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	repoFilePermPrivateRW os.FileMode = 0600
+	repoDirPermPrivateRWX os.FileMode = 0700
+)
+
+type Repo struct {
+	fileRepoDir         string
+	fileSuffix          string
+	transientFileMarker string
+
+	mu sync.Mutex
```

Comment:
nit: grouping the lock with the fields it protects is a useful convention. We generally expect to see the lock declared and the fields is protects just below it.